core/src/main/java/nu/yona/server/analysis/entities/Activity.java

```diff
@@ -0,0 +1,74 @@
+package nu.yona.server.analysis.entities;
+
+import java.util.Date;
+import java.util.UUID;
+
+import javax.persistence.Entity;
+import javax.persistence.Table;
+
+import nu.yona.server.entities.EntityWithID;
+import nu.yona.server.entities.RepositoryProvider;
+import nu.yona.server.goals.entities.Goal;
+
+@Entity
+@Table(name = "ACTIVITIES")
+public class Activity extends EntityWithID
+{
+	public static ActivityRepository getRepository()
+	{
+		return (ActivityRepository) RepositoryProvider.getRepository(Activity.class, UUID.class);
+	}
+
+	private UUID userAnonymizedID;
+	private UUID goalID;
```

Comment:
This change is already enormous, so let's not combine that change. I've logged [YD-150](http://jira.yona.nu/browse/YD-150) for this.